app/src/main/java/org/listenbrainz/android/util/Log.kt

```diff
@@ -2,17 +2,24 @@ package org.listenbrainz.android.util
 
 import com.limurse.logger.Logger
 
-object Log {
-
-    fun e(message: Any?, tag: String? = null) {
-        Logger.e(tag, msg = message.toString())
-    }
-
-    fun d(message: Any?) {
-        Logger.d(msg = message.toString())
-    }
+interface Log {
```

Comment:
This is one of the funniest code implementations I have ever seen XD

I mean... there is no need of an interface here when we are gonna make the default implementation stay the same. The only reason I kept this file was because I did not want to refactor Log.d and stuff everywhere in the existing codebase. But I think we should directly use Logger.d , etc everywhere. No need to maintain this file

Reply:
Logger does file writes underneath and instantiates new thread. It makes the components un-testable when I directly use Logger.x, this interface is just hack to make things testable. Although now we have to pass log as a parameter but small giveaway for big gains.

Reply:
@07jasjeet the logger shouldn't be a part of the test suite anyway. How come the logger is affecting the tests? We mock files for testing and logger shouldn't be used there. The logger is only supposed to be used in our implementations

Reply:
Could you please mention the tests the logger directly being used is affecting?

Reply:
In the #371, I was working with JobQueue, and using logger, it just throws errors because we are using android level components in unit tests. Same with when I was implementing Timer's tests.

Reply:
@07jasjeet Makes sense when we need to test. You will have to inject the logger class independently. The other way could be to decouple the logger entirely and implement it early on. So for our use case, this is sadly not a hack but the best way to deal with this..